From 2ce9e012cac564239b798f33069f8d659021c1df Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 19 Mar 2020 15:03:09 +0200
Subject: [PATCH] [SW-0] goya2: arc: support per target override

This is needed for example by ONIX so it won't hang on non supported
functionality e.g. efuse, PCIe FW sram

Change-Id: I19e895d1cbbe79a6a70f65531be61894f0ecbd41
---

diff --git a/Makefile b/Makefile
index e9c8c05..3197f4b 100644
--- a/Makefile
+++ b/Makefile
@@ -38,7 +38,7 @@
 CFLAGS = -DELP_ARCV2
 export CFLAGS += -Os -fdata-sections -ffunction-sections
 
-ARCBOOT_COMMON_SRCS_C := $(shell ls common/*.c)
+ARCBOOT_COMMON_SRCS_C := $(shell ls common/*.c $(HL_DEV)/$(HL_TARGET).c)
 
 ARCBOOT_RAM_SRCS_C := app/arcboot_ram.c
 ARCBOOT_RAM_SRCS_S := app/crt0_ram.S
diff --git a/common/efuse.c b/common/efuse.c
index be140dd..c438b6c 100644
--- a/common/efuse.c
+++ b/common/efuse.c
@@ -50,7 +50,7 @@
 	return 0;
 }
 
-int fuse_read(uint32_t word, uint32_t *val)
+__attribute__((weak)) int fuse_read(uint32_t word, uint32_t *val)
 {
 	int offset;
 
diff --git a/common/mem_cpy.c b/common/mem_cpy.c
index 3bf0406..16e3d51 100644
--- a/common/mem_cpy.c
+++ b/common/mem_cpy.c
@@ -51,7 +51,7 @@
 	return mem_cpy_status;
 }
 
-void mem_cpy_wait_finish(void)
+__attribute__((weak)) void mem_cpy_wait_finish(void)
 {
 	uint32_t mem_cpy_status;
 
diff --git a/goya2_arc/asic.c b/goya2_arc/asic.c
new file mode 100644
index 0000000..3f6ceaa
--- /dev/null
+++ b/goya2_arc/asic.c
@@ -0,0 +1,6 @@
+/*
+ * Copyright (C) 2017 HabanaLabs Ltd.
+ *   Author: Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
diff --git a/goya2_arc/onix.c b/goya2_arc/onix.c
new file mode 100644
index 0000000..211e27b
--- /dev/null
+++ b/goya2_arc/onix.c
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2017 HabanaLabs Ltd.
+ *   Author: Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
+
+#include <unistd.h>
+#include "arcboot.h"
+#include "efuse.h"
+
+#define EFUSE_SYSCFG_ONIX_DEFAULT \
+				(EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI | \
+				 EFUSE_SYSCFG_KEY_LOCATION_SPI | \
+				 EFUSE_SYSCFG_REVOKATION_LOCATION_SPI)
+
+int fuse_read(uint32_t word, uint32_t *val)
+{
+	int ret = -1;
+
+	if (word == EFUSE_SYSCFG_OFFSET) {
+		*val = EFUSE_SYSCFG_ONIX_DEFAULT;
+		ret = 0;
+	}
+
+	return ret;
+}
+
+void mem_cpy_wait_finish(void) {}
diff --git a/goya2_arc/pldm.c b/goya2_arc/pldm.c
new file mode 100644
index 0000000..3f6ceaa
--- /dev/null
+++ b/goya2_arc/pldm.c
@@ -0,0 +1,6 @@
+/*
+ * Copyright (C) 2017 HabanaLabs Ltd.
+ *   Author: Noam Camus <noamikong@gmail.com>
+ *
+ * SPDX-License-Identifier:     GPL-2.0+
+ */
